pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -300,7 +300,8 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 			machineDeploymentLabels[k] = v
 		}
 		machineSpecAnnotations := map[string]string{}
-		if !machinePool.DrainBeforeDelete {
+		// Ignore drain before delete etcd nodes
```

Comment:
hmm, the comment is a bit confusing, could you rephrase it to be more clear? 
```suggestion
		// Ignore drain if DrainBeforeDelete is unset or the pool is for etcd nodes 
```